infra/ballista/build/lambda-handler.py

```diff
@@ -0,0 +1,190 @@
+import base64
+import os
+import subprocess
+from subprocess import PIPE
+import logging
+import pexpect
+from pexpect import popen_spawn
+import time
+import sys
+
+
+logging.getLogger().setLevel(logging.INFO)
+
+# Create global variables for the forked processes
+process_s = None
+process_e = None
+process_cli = None
+IS_COLD_START = True
+
+
+def init():
+    # start scheduler
+    global process_s
+    process_s = subprocess.Popen(
+        ["/opt/ballista/ballista-scheduler", "--sled-dir", "/tmp/scheduler/sled"],
+        stdout=PIPE,
+        stderr=sys.stderr,
+        bufsize=0,
+    )
+    logging.info("scheduler starts")
+    time.sleep(1)
+    # wait till the scheduler is up and running
+    if process_s.poll() is None:
+        return_code = 1
+        timeout = 0
+        while return_code:
+            res = subprocess.run(
+                ["nc", "-z", "localhost", "50050"], capture_output=True
+            )
+            return_code = res.returncode
+            logging.debug(res)
+            time.sleep(1)
+            # rc = process_s.poll()
+            timeout += 1
+            if timeout >= 30:  # or rc is not None:
```

Comment:
❓ why use a timeout in number of iterations and not time? I have no idea if an iteration is long or not, so I don't know if your timeout is a lot or not. Time is absolute and easily interpreted.